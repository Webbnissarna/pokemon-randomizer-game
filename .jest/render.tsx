import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
