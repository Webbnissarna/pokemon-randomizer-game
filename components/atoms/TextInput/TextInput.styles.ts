import { ThemeUIStyleObject } from "theme-ui";

export const mainStyle: ThemeUIStyleObject = {
  borderRadius: "sm",
  border: "0px",
  backgroundColor: "inputBackground",
  color: "inputText",
  fontSize: "md",
  fontFamily: "inherit",
  height: "sm",
  paddingX: "md",

  "&:focus-visible, &:focus": {
    outline: "1px solid #999",
  },
};
