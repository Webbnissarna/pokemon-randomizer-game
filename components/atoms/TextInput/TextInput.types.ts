export type OnChangeEvent = React.ChangeEvent<HTMLInputElement>;

export interface TextInputProps {
  value: string;
  onChange: (event: OnChangeEvent) => void;
  placeholder: string;
  disabled?: boolean;
}
