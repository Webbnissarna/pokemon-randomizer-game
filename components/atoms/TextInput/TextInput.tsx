import { Input } from "theme-ui";
import { mainStyle } from "./TextInput.styles";
import type { TextInputProps } from "./TextInput.types";

export default function TextInput({
  value,
  onChange,
  placeholder,
  disabled,
}: TextInputProps): JSX.Element {
  return (
    <>
      <Input
        aria-label={placeholder}
        type="text"
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        disabled={disabled}
        sx={mainStyle}
      />
    </>
  );
}
