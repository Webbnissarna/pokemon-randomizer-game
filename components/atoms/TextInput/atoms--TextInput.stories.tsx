import { useState } from "react";
import TextInput from ".";

export function Default() {
  const [text, setText] = useState("Hello");

  return (
    <div style={{ display: "flex", flexDirection: "column", gap: 20 }}>
      <h1>Normal</h1>
      <TextInput
        value={text}
        onChange={(event) => {
          setText(event.currentTarget.value);
        }}
        placeholder="Normal input field"
      />

      <h1>Placeholder</h1>
      <TextInput
        value={""}
        onChange={() => undefined}
        placeholder="Placeholder text"
      />

      <h1>Disabled</h1>
      <TextInput
        disabled={true}
        value={text}
        onChange={() => undefined}
        placeholder="Disabled field"
      />
    </div>
  );
}
