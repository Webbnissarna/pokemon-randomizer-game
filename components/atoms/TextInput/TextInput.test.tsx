import { screen } from "@testing-library/react";
import { render } from "../../../.jest/render";
import userEvent from "@testing-library/user-event";
import TextInput from ".";

describe("Text Input", () => {
  it("renders", () => {
    const { asFragment } = render(
      <TextInput value="" placeholder="" onChange={() => undefined} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("has textbox role", () => {
    render(<TextInput value="" placeholder="" onChange={() => undefined} />);

    const element = screen.getByRole("textbox");

    expect(element).toBeTruthy();
  });

  it("shows text", () => {
    const text = "Hello World";

    render(
      <TextInput value={text} placeholder="" onChange={() => undefined} />
    );

    const element = screen.getByDisplayValue(text);

    expect(element).toBeTruthy();
  });

  it("shows placeholder", () => {
    const placeholderText = "Placeholder text";

    render(
      <TextInput
        value=""
        placeholder={placeholderText}
        onChange={() => undefined}
      />
    );

    const element = screen.getByPlaceholderText(placeholderText);

    expect(element).toBeTruthy();
  });

  it("calls onChange", async () => {
    const text = "Hello world";
    const callback = jest.fn();
    const user = userEvent.setup();

    render(<TextInput value="" placeholder="" onChange={callback} />);

    await user.type(screen.getByRole("textbox"), text);

    expect(callback).toHaveBeenCalledTimes(text.length);
  });
});
