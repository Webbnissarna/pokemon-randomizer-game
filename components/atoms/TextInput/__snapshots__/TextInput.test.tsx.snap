// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <input
    aria-label=""
    class="css-kdrqma-Box"
    placeholder=""
    type="text"
    value=""
  />
</DocumentFragment>
`;
